engine/apps/alerts/tasks/notify_all.py

```diff
@@ -18,8 +18,24 @@ def notify_all_task(alert_group_pk, escalation_policy_snapshot_order=None):
 
     alert_group = AlertGroup.all_objects.get(pk=alert_group_pk)
 
+    # check alert group state before notifying all users in the channel
+    if alert_group.resolved or alert_group.acknowledged or alert_group.silenced:
+        task_logger.info(f"alert_group {alert_group.pk} was resolved, acked or silenced forever. No need to notify all")
+        return
+
     escalation_snapshot = alert_group.escalation_snapshot
-    escalation_policy_snapshot = escalation_snapshot.escalation_policies_snapshots[escalation_policy_snapshot_order]
+    escalation_policy_snapshot = escalation_snapshot.escalation_policies_snapshots.get(escalation_policy_snapshot_order)
```

Comment:
Isn't `escalation_policies_snapshots` a list? Seems like using `get` here won't work

Reply:
You are right, thank you, I'll fix it